low-level/sensors/impl/generic/node_hw.py

```diff
@@ -480,6 +481,63 @@ def run(self):
             logger.warning(f"{self.SENSOR_NAME} Node hw monitoring disabled")
             self.shutdown()
 
+    def _check_faulty_resource_status(self):
+        # Currently, the SSPL sensor relies on IPMI SEL events to track server
+        # resources fault & recovery. However, its been observed multiple times
+        # with Supermicro servers, that SEL not getting updated in case recoveries
+        # of some resources.
+        # here we are polling resources that are detected to be faulty or missing
+        # periodically, in addition to current polling for SEL.
+
+        # Copying object to avoid RuntimeError: dictionary changed size during iteration
```

Comment:
need to have it under try - exception block, there are chances of exception here, include _generate_resource_alert() call

Reply:
Done.
Addressed this review comments in https://github.com/Seagate/cortx-monitor/pull/393